lib/reducers/create-otp-reducer.js

```diff
@@ -129,22 +147,35 @@ function createOtpReducer (config, initialQuery) {
         return update(state, { currentQuery: { [action.payload.type]: { $set: action.payload.location } } })
       case 'CLEAR_LOCATION':
         return update(state, { currentQuery: { [action.payload.type]: { $set: null } } })
-      case 'SET_MODE':
-        return update(state, { currentQuery: { mode: { $set: action.payload.mode } } })
-      case 'SET_DEPART':
-        return update(state, { currentQuery: { departArrive: { $set: action.payload.departArrive } } })
-      case 'SET_DATE':
-        return update(state, { currentQuery: { date: { $set: action.payload.date } } })
-      case 'SET_TIME':
-        return update(state, { currentQuery: { time: { $set: action.payload.time } } })
+
+      case 'SET_QUERY_PARAM':
+        console.log('merging QPs', action.payload)
+        return update(state, { currentQuery: { $merge: action.payload } })
```

Comment:
It could also be possible for a user to send an action with this name.  Perhaps we should namespace all of these actions so that there is less of a chance for collisions with redux reducers implemented in upstream apps.